reactioncommerce/reaction
# Add updateOrderFulfillmentGroup mutation

Part of #4999
Impact: **minor**  
Type: **feature**

## Changes
A new `updateOrderFulfillmentGroup` mutation allows you to change the `tracking` and `status` fields of an order fulfillment group after the order has been placed.

The operator UI now uses this mutation for the `tracking` update only. Changing the operator UI to use GraphQL for group status updates will be done in a later PR.

- `updateOrderFulfillmentGroup` can be called internally to synchronize from an external system by setting `context.isInternalCall` to `true`
- `updateOrderFulfillmentGroup` can be called through GraphQL by any user with "orders" permission for the shop that owns the order
- The "orders/updateShipmentTracking" Meteor method is removed in favor of the `updateOrderFulfillmentGroup` GraphQL mutation

## Breaking changes
None

## Testing
1. Place an order.
1. Verify that the mutation works as described.
1. Verify that you can update the order fulfillment group status and tracking when authenticated as a user with "orders" role, but not when authenticated as someone else or when not authenticated.
1. Verify that saving the Tracking field in the operator UI works.
